src/data-access/repositories/mongo/mongo-device-repository.ts

```diff
@@ -1,24 +1,56 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
 import { BaseMongoRepository } from './base-mongo-repository'
 import { DeviceRepository } from '../interfaces/device-repository'
 import { Device } from '../../../model/entities/device'
 import { MongoDatabase } from './mongo-database'
-import { MongoDevice, MongoEntityConverter } from './mongo-entity-converter'
+import { INewsDevice } from '../../../model/entities/inews-device'
+import { TelemetricsDevice } from '../../../model/entities/telemetrics-device'
 
-const DEVICE_COLLECTION_NAME: string = 'peripheralDevices'
+const DEVICE_COLLECTION_NAME: string = 'externalDevices'
 
 export class MongoDeviceRepository extends BaseMongoRepository implements DeviceRepository {
 
-  constructor(mongoDatabase: MongoDatabase, private readonly mongoEntityConverter: MongoEntityConverter) {
+  constructor(mongoDatabase: MongoDatabase) {
     super(mongoDatabase)
   }
 
   protected getCollectionName(): string {
     return DEVICE_COLLECTION_NAME
   }
 
-  public async getDevices(): Promise<Device[]> {
-    this.assertDatabaseConnection(MongoDeviceRepository.prototype.getDevices.name)
-    const mongoDevices: MongoDevice[] = await this.getCollection().find<MongoDevice>({}).toArray()
-    return this.mongoEntityConverter.convertToDevices(mongoDevices)
+  public async findAllDevices(): Promise<(Device | INewsDevice | TelemetricsDevice)[]> {
+    this.assertDatabaseConnection(MongoDeviceRepository.prototype.findAllDevices.name)
+    const devices: Device[] = await this.getCollection().find<(Device | INewsDevice | TelemetricsDevice)>({}).toArray() 
+    return devices
+  }
```

Comment:
You can change the type to simply be `Promise<Device[]>`
And then you can return the result directly, so it become:
```
public async findAllDevices(): Promise<Device>[] {
  ... // the assertion here
  return await this.getCollection().find()... // etc...
}
```